db/versions/0017.yml

```diff
@@ -0,0 +1,166 @@
+        cond text[];
+        exp_cond_operator text;
+        exp_cond_operand timestamptz;
+        partition_key_var text;
+        row_key_var text;
+      begin
+        if not condition is null then
+          raise exception 'no condition support in denylisted_notification_entities_scan';
+        else
+          return query select
+            encode_string_key(denylisted_notification.notification_type) as partition_key,
+            encode_string_key(denylisted_notification.notification_address) as row_key,
+            jsonb_build_object(
+              'PartitionKey', encode_string_key(denylisted_notification.notification_type),
+              'RowKey', encode_string_key(denylisted_notification.notification_address),
+              'notificationType', notification_type,
+              'notificationAddress', notification_address
+            ) as value,
+            1 as version,
+            denylisted_notification.etag as etag from denylisted_notification
+          where
+            (denylisted_notification_entities_scan.pk is null or decode_string_key(denylisted_notification_entities_scan.pk) = notification_type) and
+            (denylisted_notification_entities_scan.rk is null or decode_string_key(denylisted_notification_entities_scan.rk) = notification_address)
+          order by denylisted_notification.notification_type, denylisted_notification.notification_address
+          limit case
+            when (size is not null and size > 0) then size + 1
+            else null
+          end
+          offset case
+            when (size is not null and size > 0 and page is not null and page > 0) then page
+            else 0
+          end;
+        end if;
+      end
+  add_denylist_address:
+    description: |-
+      If the denylist address already exists, this is a no-op. Otherwise, add the denylist
+      address for the taskcluster-notify service, with a new random etag.
+    mode: write
+    serviceName: notify
+    args: address_in jsonb
+    returns: void
+    body: |-
+      begin
+        insert into denylisted_notification(notification_type, notification_address, etag)
+        values (
+          (address_in ->> 'notificationType')::text,
+          (address_in ->> 'notificationAddress')::text,
+          public.gen_random_uuid()
+        ) on conflict do nothing;
+      end
+  delete_denylist_address:
+    description: |-
+      Delete a denylist address for the taskcluster-notify service.
+      Returns number of rows deleted (0 or 1).
+    mode: write
+    serviceName: notify
+    args: address_in jsonb
+    returns: integer
+    body: |-
+      begin
+        delete from denylisted_notification where
+        denylisted_notification.notification_type = (address_in ->> 'notificationType')::text and
+        denylisted_notification.notification_address = (address_in ->> 'notificationAddress')::text;
+        if found then
+          return 1;
+        end if;
+        return 0;
+      end
+  all_denylist_address:
```

Comment:
The singular here reads awkwardly to me.  Maybe pluralize it?

Reply:
ok